fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/EliminateAggRule.java

```diff
@@ -80,63 +80,54 @@ public static EliminateAggRule getInstance() {
 
     private static final EliminateAggRule INSTANCE = new EliminateAggRule();
 
+    private static final Set<String> SUPPORTED_AGG_FUNCTIONS = ImmutableSet.of(
+            FunctionSet.SUM, FunctionSet.COUNT, FunctionSet.AVG, FunctionSet.FIRST_VALUE,
+            FunctionSet.MAX, FunctionSet.MIN, FunctionSet.GROUP_CONCAT
+    );
+
     @Override
     public boolean check(OptExpression input, OptimizerContext context) {
-        LogicalAggregationOperator aggOp = input.getOp().cast();
-        List<ColumnRefOperator> groupKeys = aggOp.getGroupingKeys();
-
-        for (Map.Entry<ColumnRefOperator, CallOperator> entry : aggOp.getAggregations().entrySet()) {
-            if (entry.getValue().isDistinct()) {
-                return false;
-            }
-            String fnName = entry.getValue().getFnName();
-            if (!(fnName.equals(FunctionSet.SUM) || fnName.equals(FunctionSet.COUNT) ||
-                    fnName.equals(FunctionSet.AVG) ||
-                    fnName.equals(FunctionSet.FIRST_VALUE) ||
-                    fnName.equals(FunctionSet.MAX) || fnName.equals(FunctionSet.MIN) ||
-                    fnName.equals(FunctionSet.GROUP_CONCAT))) {
-                return false;
-            }
+        if (!context.getSessionVariable().isEnableUKFKOpt()) {
```

Comment:
This is a behavior change? maybe we set `enable_ukfk_opt = true` by default later?

Reply:
Oh, this make a behavior change. Remove this check.

Reply:
Add a new session variable `enable_eliminate_agg` (default is true) to avoid behaviour changed.